src/zenml/integrations/hyperai/service_connectors/hyperai_service_connector.py

```diff
@@ -0,0 +1,405 @@
+        """Get the default resource ID for a resource type.
+
+        Args:
+            resource_type: The type of the resource to get a default resource ID
+                for. Only called with resource types that do not support
+                multiple instances.
+
+        Returns:
+            The default resource ID for the resource type.
+        """
+        instance_name = self.config.instance_name
+        if instance_name is None:
+            instance_name = "Unnamed instance"
+
+        ip_address = str(self.config.ip_address).strip()
+
+        return f"{instance_name} (IP {ip_address})"
+
+    def _paramiko_key_type_given_auth_method(self) -> paramiko.PKey:
+        """Get the Paramiko key type given the authentication method.
+
+        Returns:
+            The Paramiko key type.
+        """
+        mapping = {
+            HyperAIAuthenticationMethods.RSA_KEY_OPTIONAL_PASSPHRASE: paramiko.RSAKey,
+            HyperAIAuthenticationMethods.DSA_KEY_OPTIONAL_PASSPHRASE: paramiko.DSSKey,
+            HyperAIAuthenticationMethods.ECDSA_KEY_OPTIONAL_PASSPHRASE: paramiko.ECDSAKey,
+            HyperAIAuthenticationMethods.ED25519_KEY_OPTIONAL_PASSPHRASE: paramiko.Ed25519Key,
+        }
+
+        try:
+            return mapping[self.auth_method]
+        except KeyError:
+            raise ValueError(
+                f"Invalid authentication method: {self.auth_method}"
+            )
+
+    def _authorize_client(self) -> None:
+        """Verify that the client can authenticate with the HyperAI instance.
+
+        Raises:
+            BadHostKeyException: If the host key is invalid.
+            AuthenticationException: If the authentication credentials are
+                invalid.
+            SSHException: If there is an SSH related error.
+        """
+        logger.info("Verifying connection to HyperAI instance...")
+
+        if self.config.ssh_passphrase is None:
+            ssh_passphrase = None
+        else:
+            ssh_passphrase = self.config.ssh_passphrase.get_secret_value()
+
+        # Connect to the HyperAI instance
+        try:
+            # Convert the SSH key from base64 to string
+            base64_key_value = self.config.base64_ssh_key.get_secret_value()
+            ssh_key = base64.b64decode(base64_key_value).decode("utf-8")
+            paramiko_key = None
+
+            # Create temporary file and write Docker Compose file to it
+            with tempfile.NamedTemporaryFile(mode="w", delete=True) as f:
+                # Get file path
+                file_path = f.name
+
+                # Write Docker Compose file to temporary file
+                with f.file as f_:
+                    f_.write(ssh_key)
+
+                paramiko_key = self._paramiko_key_type_given_auth_method().from_private_key_file(
+                    file_path, password=ssh_passphrase
+                )
+
+            # Trim whitespace from the IP address
+            ip_address = str(self.config.ip_address).strip()
```

Comment:
This is already a string so I think we don't need the conversion?

Reply:
Done